integration/spark-datasource/src/main/spark2.3/org/apache/spark/sql/CarbonVectorProxy.java

```diff
@@ -0,0 +1,276 @@
+
+    public void putBoolean(int rowId, boolean value, int ordinal) {
+        columnVectors[ordinal].putBoolean(rowId, (boolean) value);
+    }
+
+    public void putByte(int rowId, byte value, int ordinal) {
+        columnVectors[ordinal].putByte(rowId, (byte) value);
+    }
+
+    public void putShort(int rowId, short value, int ordinal) {
+        columnVectors[ordinal].putShort(rowId, (short) value);
+    }
+
+    public void putInt(int rowId, int value, int ordinal) {
+        columnVectors[ordinal].putInt(rowId, (int) value);
+    }
+
+    public void putDictionaryInt(int rowId, int value, int ordinal) {
+        columnVectors[ordinal].getDictionaryIds().putInt(rowId, (int) value);
+    }
+
+    public void putFloat(int rowId, float value, int ordinal) {
+        columnVectors[ordinal].putFloat(rowId, (float) value);
+    }
+
+    public void putLong(int rowId, long value, int ordinal) {
+        columnVectors[ordinal].putLong(rowId, (long) value);
+    }
+
+    public void putDouble(int rowId, double value, int ordinal) {
+        columnVectors[ordinal].putDouble(rowId, (double) value);
+    }
+
+    public void putByteArray(int rowId, byte[] value, int ordinal) {
+        columnVectors[ordinal].putByteArray(rowId, (byte[]) value);
+    }
+
+    public void putInts(int rowId, int count, int value, int ordinal) {
+        columnVectors[ordinal].putInts(rowId, count, value);
+    }
+
+    public void putShorts(int rowId, int count, short value, int ordinal) {
+        columnVectors[ordinal].putShorts(rowId, count, value);
+    }
+
+    public void putLongs(int rowId, int count, long value, int ordinal) {
+        columnVectors[ordinal].putLongs(rowId, count, value);
+    }
+
+    public void putDecimal(int rowId, Decimal value, int precision, int ordinal) {
+        columnVectors[ordinal].putDecimal(rowId, value, precision);
+
+    }
+
+    public void putDoubles(int rowId, int count, double value, int ordinal) {
+        columnVectors[ordinal].putDoubles(rowId, count, value);
+    }
+
+    public void putByteArray(int rowId, byte[] value, int offset, int length, int ordinal) {
+        columnVectors[ordinal].putByteArray(rowId, (byte[]) value, offset, length);
+    }
+
+    public void putNull(int rowId, int ordinal) {
+        columnVectors[ordinal].putNull(rowId);
+    }
+
+    public void putNulls(int rowId, int count, int ordinal) {
+        columnVectors[ordinal].putNulls(rowId, count);
+    }
+
+    public void putNotNull(int rowId, int ordinal) {
+        columnVectors[ordinal].putNotNull(rowId);
+    }
+
+    public void putNotNulls(int rowId, int count, int ordinal) {
+        columnVectors[ordinal].putNotNulls(rowId, count);
+    }
+
+    public boolean isNullAt(int rowId, int ordinal) {
+        return columnVectors[ordinal].isNullAt(rowId);
+    }
+
+    public boolean hasDictionary(int ordinal) {
+        return columnVectors[ordinal].hasDictionary();
+    }
+
+    public void setDictionary(Object dictionary, int ordinal) {
```

Comment:
Please make sure we set correct dictionary. 2.3 does not  expect org.apache.parquet.column.Dictionary

Reply:
Yes, proxy layer corresponding to the spark 2.3 and spark 2.1&2.2 is setting the dictionary instance